lib/pymice/_ICData.py

```diff
@@ -1029,6 +977,63 @@ def _getCageCornerSide(self, Cage, Corner, Side):
 
     return cage, corner, corner[Side]
 
+  @staticmethod
+  def _findAndOpenZipFile(zf, path):
+    try:
+      return zf.open(path)
+
+    except KeyError:
+      return zf.open('IntelliCage/' + path)
+
+  def extractSessions(self, zf):
+    try:
+      with self._findAndOpenZipFile(zf, 'Sessions.xml') as fh:
+        dom = minidom.parse(fh)
+
+      aos = dom.getElementsByTagName('ArrayOfSession')[0]
+      ss = aos.getElementsByTagName('Session')
+      sessions = []
+      for session in ss:
+        # offset = session.getElementsByTagName('TimeZoneOffset')[0]
+        # offset = offset.childNodes[0]
+        # assert offset.nodeType == offset.TEXT_NODE
+        # offset = offset.nodeValue
+
+        interval = session.getElementsByTagName('Interval')[0]
+        start = interval.getElementsByTagName('Start')[0]
+        start = start.childNodes[0]
+        assert start.nodeType == start.TEXT_NODE
+        start = dateutil.parser.parse(start.nodeValue)
+
+        end = interval.getElementsByTagName('End')[0]
+        end = end.childNodes[0]
+        assert end.nodeType == end.TEXT_NODE
+        end = end.nodeValue
+        end = None if end.startswith('0001') else dateutil.parser.parse(end)
+
+        if end is not None and start.tzinfo != end.tzinfo:
+          warn.warn(UserWarning('Timezone changed!'))
+
+        for sessionStart, sessionEnd in sessions:
+          if sessionEnd is None:
+            continue
+
+          if sessionStart < start < sessionEnd or \
+                  end is not None and sessionStart < end < sessionEnd or \
+                  (end is not None and start <= sessionStart and sessionEnd <= end) or \
+                  (end is not None and sessionStart <= start and end <= sessionEnd):
+            warn.warn(UserWarning('Temporal overlap of sessions!'))
+
+        sessions.append(Session(Start=start, End=end))
+
+      sessions = sorted(sessions, key=attrgetter('Start'))
+
+    except:
+      sessions = None
+      pass
+
+    return sessions
```

Comment:
There is a lot happening in this function. Could you divide it into two separate functions: one that parses data and the other that evaluates parsed data and warns the user.